lib/matplotlib/cbook.py

```diff
@@ -2385,7 +2384,7 @@ def pts_to_midstep(x, *args):
     # do normalization
     vertices = _step_validation(x, *args)
     # create the output array
-    steps = ma.zeros((vertices.shape[0], 2 * len(x)), np.float)
+    steps = np.zeros((vertices.shape[0], 2 * len(x)), np.float)
```

Comment:
Before this patch, `pts_to_prestep` was using `np.zeros` whereas `pts_to_midstep` and `pts_to_poststep` were using `ma.zeros`, so the behavior was inconsistent to start with.  I guess that the use of `ma.zeros` was to handle the hypothetical case where a masked array would be passed to a step-plotting function, but `Line2D.recache` (which ultimately dispatches to `pts_to_*step`) already replaces masked arrays by nan-filled regular arrays, so there's no reason to return a masked array here.